pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -421,22 +421,15 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 		cr.Status.LastSyncGeneration = origCR.Generation
 	}
 
+	// TODO: if syncErr is not nil, and this updateStatus runs, we hotloop.
+	// If we remove the updateStatus call, we do not hotloop.
 	err = r.updateStatus(origCR, cr, logger)
```

Comment:
@joelddiaz @csrwng weirdest thing i've seen yet... ^^^

Reply:
Answer was right infront of my face, Cesar pointed out on slack, it's because the update status is seeing a change, and then re-syncing the cluster every error, aka our likely hotloop that was eating all AWS requests.